superset-frontend/src/common/hooks/apiResources/dashboards.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { useApiV1Resource } from './apiResources';
+
+export const useDashboard = (idOrSlug: string | number) =>
+  useApiV1Resource(`/api/v1/dashboard/${idOrSlug}`);
```

Comment:
cc @eschutho just to raise awareness of pattern use

Reply:
Thanks. I guess we'll continue to use component state for now instead of redux until we come to a final decision on the pattern? Obviously we don't want to hold up this SPA effort, but it may mean more refactoring later on.

Reply:
I think we should also start to look at third party libs that have a very similar API to the `useResource` hooks. @ktmud linked some good ones in the original PR: #13218